creator-node/src/services/stateMachineManager/CNodeHealthManager.ts

```diff
@@ -0,0 +1,318 @@
+  _computeContentNodePeerSet
+}
+
+/**
+ * Performs a health check on the peer set
+ * @param {Object[]} nodeUsers array of objects of schema { primary, secondary1, secondary2, user_id, wallet }
+ * @param {string} contentNodeEndpoint the IP address / URL of this Content Node
+ * @param {boolean?} [performSimpleCheck=false] flag to dictate whether or not to check health check response to
+ *  determine node health
+ * @returns the unhealthy peers in a Set
+ *
+ * @note consider returning healthy set?
+ * TODO - add retry logic to node requests
+ */
+async function getUnhealthyPeers(
+  nodeUsers: StateMonitoringUser[],
+  thisContentNodeEndpoint: string,
+  performSimpleCheck = false
+) {
+  // Compute content node peerset from nodeUsers (all nodes that are in a shared replica set with this node)
+  const peerSet = CNodeHealthManager._computeContentNodePeerSet(
+    nodeUsers,
+    thisContentNodeEndpoint
+  )
+
+  /**
+   * Determine health for every peer & build list of unhealthy peers
+   * TODO: change from sequential to chunked parallel
+   */
+  const unhealthyPeers = new Set()
+
+  for await (const peer of peerSet) {
+    const isHealthy = await CNodeHealthManager.isNodeHealthy(
+      peer,
+      performSimpleCheck
+    )
+    if (!isHealthy) {
+      unhealthyPeers.add(peer)
+    }
+  }
+
+  return unhealthyPeers
+}
+
+async function isNodeHealthy(node: string, performSimpleCheck = false) {
+  try {
+    const verboseHealthCheckResp =
+      await CNodeHealthManager._queryVerboseHealthCheck(node)
+    // if node returns healthy: false consider that unhealthy just like non-200 response
+    const { healthy } = verboseHealthCheckResp
+    if (!healthy) {
+      throw new Error(`Node health check returned healthy: false`)
+    }
+    if (!performSimpleCheck) {
+      CNodeHealthManager.determinePeerHealth(verboseHealthCheckResp)
+    }
+  } catch (e: any) {
+    logger.error(`isNodeHealthy() peer=${node} is unhealthy: ${e.toString()}`)
+    return false
+  }
+
+  return true
+}
+
+/**
+ * Takes data off the verbose health check response and determines the peer heatlh
+ * @param {Object} verboseHealthCheckResp verbose health check response
+ *
+ * TODO: consolidate CreatorNodeSelection + peer set health check calculation logic
+ */
+function determinePeerHealth(verboseHealthCheckResp: any) {
+  // Check for sufficient minimum storage size
+  const { storagePathSize, storagePathUsed } = verboseHealthCheckResp
+  if (
+    storagePathSize &&
+    storagePathUsed &&
+    !hasEnoughStorageSpace({
+      storagePathSize,
+      storagePathUsed,
+      maxStorageUsedPercent: MAX_STORAGE_USED_PERCENT
+    })
+  ) {
+    throw new Error(
+      `Almost out of storage=${
+        storagePathSize - storagePathUsed
+      }bytes remaining out of ${storagePathSize}. Requires less than ${MAX_STORAGE_USED_PERCENT}% used`
+    )
+  }
+
+  // Check for sufficient memory space
+  const { usedMemory, totalMemory } = verboseHealthCheckResp
```

Comment:
i think we should remove the memory check here. thoughts @SidSethi?

Reply:
done by Johannes [here](https://github.com/AudiusProject/audius-protocol/commit/ebf9961185cae91774e721681b32fe8ee6c75a13) in the JS version, which is deleted. updated TS version to delete it in this PR as well